mh/database.py

```diff
@@ -886,7 +886,190 @@ def delete_friend(self, capcom_id, friend_id):
         return self.parent.delete_friend(capcom_id, friend_id)
 
 
-# TODO: Backport MySQLDatabase
+class MySQLDatabase(TempDatabase):
+    """Hybrid MySQL/TempDatabase.
+
+    Requires mysql-connector-python:
+     - 8.0.23 (latest Python2 version)
+     - Python3 users can pick more recent versions
+
+    TODO:
+     - Add reconnect wrapper
+     - Backport methods used by central/state if needed
+    """
+
+    def __init__(self):
+        self.parent = super(MySQLDatabase, self)
+        self.parent.__init__()
+        from mysql import connector
+        self.connection = connector.connect(
+            **utils.get_mysql_config("MYSQL")
+        )
+        self.create_database()
+        self.populate_database()
+
+    def create_database(self):
+        with self.connection.cursor() as cursor:
+            # Consoles creation
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS consoles"
+                " (support_code VARCHAR(11) NOT NULL,"
+                " slot_index TINYINT NOT NULL,"
+                " capcom_id VARCHAR(6) NOT NULL,"
+                " name VARCHAR(8),"
+                " PRIMARY KEY(capcom_id),"
+                " UNIQUE profiles_uniq_idx (support_code, slot_index));"
+            )
+
+            # Friends creation
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS friend_lists"
+                " (id MEDIUMINT NOT NULL AUTO_INCREMENT,"
+                " capcom_id VARCHAR(6) NOT NULL,"
+                " friend_id VARCHAR(6) NOT NULL,"
+                " PRIMARY KEY (id),"
+                " UNIQUE friends_uniq_idx (capcom_id, friend_id));"
+            )
+
+    def populate_database(self):
+        with self.connection.cursor() as cursor:
+            cursor.execute("SELECT * FROM consoles")
```

Comment:
I don't quite like that we are loading the entire table into memory when not all slot may be used. After we land this, this should be refactored into a more lazy operation.